src/ifcwrap-csharp/IfcCSharp.i

```diff
@@ -0,0 +1,377 @@
+#endif
+#ifdef HAS_SCHEMA_4x3_rc4
+#include "../ifcparse/Ifc4x3_rc4.h"
+#endif
+#ifdef HAS_SCHEMA_4x3
+#include "../ifcparse/Ifc4x3.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_tc1
+#include "../ifcparse/Ifc4x3_tc1.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_add1
+#include "../ifcparse/Ifc4x3_add1.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_add2
+#include "../ifcparse/Ifc4x3_add2.h"
+#endif
+
+	#include "../ifcparse/aggregate_of_instance.h"
+	
+	#include "../ifcparse/Argument.h"
+	#include "../ifcparse/IfcBaseClass.h"
+	#include "../ifcparse/IfcFile.h"
+	#include "../ifcparse/IfcSchema.h"
+	#include "../ifcparse/utils.h"
+
+	#include "../ifcgeom/ConversionSettings.h"
+	#include "../ifcgeom/ConversionResult.h"
+
+	#include "../svgfill/src/svgfill.h"
+
+#ifdef IFOPSH_WITH_CGAL
+	#include "../ifcgeom/kernels/cgal/CgalConversionResult.h"
+#endif
+%}
+
+// Create docstrings for generated python code.
+%feature("autodoc", "1");
+
+%shared_ptr(aggregate_of_instance)
+%include "utils/aggregate_of_instance.i"
+%include "utils/typemaps_out.i"
+
+%module ifcopenshell_net %{
+	#include "../ifcgeom/Converter.h"
+	#include "../ifcgeom/taxonomy.h"
+#ifdef IFOPSH_WITH_OPENCASCADE
+	#include "../ifcgeom/Serialization/Serialization.h"
+	#include "../ifcgeom/kernels/opencascade/IfcGeomTree.h"
+
+	#include <BRepTools_ShapeSet.hxx>
+#endif
+	#include "../ifcgeom/Iterator.h"
+	#include "../ifcgeom/ConversionResult.h"
+
+	#include "../serializers/SvgSerializer.h"
+	#include "../serializers/WavefrontObjSerializer.h"
+	#include "../serializers/HdfSerializer.h"
+	#include "../serializers/XmlSerializer.h"
+	#include "../serializers/GltfSerializer.h"
+	
+#ifdef HAS_SCHEMA_2x3
+	#include "../ifcparse/Ifc2x3.h"
+#endif
+#ifdef HAS_SCHEMA_4
+	#include "../ifcparse/Ifc4.h"
+#endif
+#ifdef HAS_SCHEMA_4x1
+	#include "../ifcparse/Ifc4x1.h"
+#endif
+#ifdef HAS_SCHEMA_4x2
+	#include "../ifcparse/Ifc4x2.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_rc1
+	#include "../ifcparse/Ifc4x3_rc1.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_rc2
+	#include "../ifcparse/Ifc4x3_rc2.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_rc3
+	#include "../ifcparse/Ifc4x3_rc3.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_rc4
+	#include "../ifcparse/Ifc4x3_rc4.h"
+#endif
+#ifdef HAS_SCHEMA_4x3
+	#include "../ifcparse/Ifc4x3.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_tc1
+	#include "../ifcparse/Ifc4x3_tc1.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_add1
+	#include "../ifcparse/Ifc4x3_add1.h"
+#endif
+#ifdef HAS_SCHEMA_4x3_add2
+	#include "../ifcparse/Ifc4x3_add2.h"
+#endif
+
+	#include "../ifcparse/ArgumentType.h"
+	#include "../ifcparse/Argument.h"
+	#include "../ifcparse/aggregate_of_instance.h"
+	#include "../ifcparse/IfcBaseClass.h"
+	#include "../ifcparse/IfcFile.h"
+	#include "../ifcparse/IfcSchema.h"
+	#include "../ifcparse/utils.h"
+	
+	#include "../ifcgeom/ConversionSettings.h"
+	#include "../ifcgeom/ConversionResult.h"
+
+	#include "../svgfill/src/svgfill.h"
+%}
+
+%include "IfcGeomWrapper.i"
+%include "IfcParseWrapper.i"
+%include "std_vector.i"
+	
+%inline %{
```

Comment:
Please add a comment answering the following questions:
- What does it do?
- Why do we need it?
- Where is the swig preprocessor inserting it?